test/e2e/tests/helper.py

```diff
@@ -0,0 +1,57 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+#	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Helper functions for acmpca tests
+"""
+
+import logging
+
+class ACMPCAValidator:
+    def __init__(self, acmpca_client):
+        self.acmpca_client = acmpca_client
+
+    def assert_certificate_authority(self, ca_arn: str, status: str, exists=True):
+        try:
+            aws_res = self.acmpca_client.describe_certificate_authority(CertificateAuthorityArn=ca_arn)
+            test_ca = aws_res["CertificateAuthority"]
+            logging.info(test_ca["Status"])
+            assert test_ca["Status"] == status
+            assert test_ca["Type"] == "ROOT"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["CommonName"] == "www.example.com"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["Country"] == "US"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["Locality"] == "Arlington"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["Organization"] == "Example Organization"
+            assert test_ca["CertificateAuthorityConfiguration"]["Subject"]["State"] == "Virginia"
+            assert test_ca["CertificateAuthorityConfiguration"]["KeyAlgorithm"] == "RSA_2048"
+            assert test_ca["CertificateAuthorityConfiguration"]["SigningAlgorithm"] == "SHA256WITHRSA"
```

Comment:
From L29 to L36: correct me if i'm wrong, but this looks like it's very tied to the test case itself and those values could be different if we change the Spec of the test resource. If that's the case let's move those assertions after L73 `acmpca_validator.assert_certificate_authority(resource_arn, "PENDING_CERTIFICATE")` and keep this function only to assert that the CA was created successfully and and the status is matching.

Reply:
Also the `exists=True` parameter is not used in all the methods